client/test/e2e/Auth.test.ts

```diff
@@ -1,40 +1,40 @@
 // src: https://playwright.dev/docs/writing-tests
 
 import { test, expect } from '@playwright/test';
-import links from './Links.ts'; // Extension is required with Playwright import
+import links from './Links.ts'; // Extension is required with Playwright import - ignore VSCode warning
 
 test.describe('Tests on Authentication Flow', () => {
   test.beforeEach(async ({ page }) => {
     await page.goto('./');
-    await page.getByRole('textbox', { name: 'username' }).fill('user-test');
+    await page.getByRole('textbox', { name: 'username' }).fill('user1');
   });
 
   test('Homepage has correct title and signin link', async ({ page }) => {
     //    await page.goto(config.baseURL);
     await expect(page).toHaveTitle('The Digital Twin as a Service');
 
-    await page.locator('button:has-text("Sign In")').click();
+    await page.getByRole('button').filter({ hasText: 'Sign In' }).click();
 
     await expect(page).toHaveURL(/.*library/);
   });
 
   test('Fill the signin fields', async ({ page }) => {
-    await page.locator('input[name="username"]').fill('user@au.dk');
-    await page.locator('button').first().click();
+    await page.getByLabel('username').fill('user@au.dk');
+    await page.getByRole('button').first().click();
 
     await expect(page).toHaveURL(/.*library/);
   });
 
   test('Account Button Contents and Links', async ({ page, baseURL }) => {
-    await page.locator('button:has-text("Sign In")').click();
+    await page.getByRole('button').filter({ hasText: 'Sign In' }).click();
     await expect(page).toHaveURL(/.*library/);
 
-    await page.locator('[aria-label="Open settings"]').click();
-    await page.locator('text=Account').click();
+    await page.getByLabel('Open settings').click();
+    await page.getByText('Account').click();
     await expect(page).toHaveURL('./account');
 
-    await page.locator('[aria-label="Open settings"]').click();
-    await page.locator('text=Logout').click();
+    await page.getByLabel('Open settings').click();
```

Comment:
Both are filtering down to the same element. I see the advantage of readability. Is there any other advantage beyond readability?

Reply:
It is also possible to add filtering options to the query. I strongly prefer this, as it's 1-1 with jest queries.
```ts
getByLabel('Open settings', {name: "myDiv"})
```